src/downloader.ts

```diff
@@ -0,0 +1,99 @@
+import { execSync } from 'child_process'
+import { unzip } from 'cross-zip'
+import { ensureDir, existsSync, writeFileSync } from 'fs-extra'
+import fetch from 'node-fetch'
+import { arch, platform } from 'os'
+import { parse } from 'path'
+import { promisify } from 'util'
+import { logger } from './logger'
+import { getPath, paths } from './path'
+
+const unzipAsync = promisify(unzip)
+const thisPlatform = platform()
+const thisArch = arch()
+
+export async function runDownloader(): Promise<void> {
+  await ensureDir(paths.data)
+  await ensureAsset(
+    'https://github.com/ethersphere/bee/releases/download/v1.5.1/bee-darwin-amd64',
+    'bee',
+    'darwin',
+    'x64',
+    true,
+  )
+  await ensureAsset(
+    'https://github.com/ethersphere/bee/releases/download/v1.5.1/bee-darwin-arm64',
+    'bee',
+    'darwin',
+    'arm64',
+    true,
+  )
+  await ensureAsset(
+    'https://github.com/ethersphere/bee/releases/download/v1.5.1/bee-linux-amd64',
+    'bee',
+    'linux',
+    'x64',
+    true,
+  )
+  await ensureAsset(
+    'https://github.com/ethersphere/bee/releases/download/v1.5.1/bee-windows-amd64.exe',
+    'bee.exe',
+    'win32',
+    'x64',
+  )
```

Comment:
I would suggest to put the platform check here as a `switch` statement. It would make it more readable.

Reply:
the architecture could then be an if statement.